### PR TITLE
SOLR-16730: Exclude username, roles and permissions for inter-node requests to SystemInfoHandler

### PR DESCRIPTION
**Pull request by tflobbe**
This PR addresses `SOLR-16730` by completely skipping including the username, roles and permissions when the request is an inter-node request to `SystemInfoHandler`. 
Having this part alone:
```
if (roles == null) {
     info.add("permissions", Set.of());
}
```
should be enough to handle the NPE described in the Jira issue, but I think it may be better to just skip this section for internal requests, since the data displayed is unrelated to the user that issues the request.

### CODE REVIEW

**Comment by janhoy** [comment]
According to RBAP api, the `[getUserRoles()](https://github.com/apache/solr/blob/main/solr/core/src/java/org/apache/solr/security/RuleBasedAuthorizationPluginBase.java#L371)` call should not return null but empty set. So I think that can be fixed too, by making [this](https://github.com/apache/solr/blob/main/solr/core/src/java/org/apache/solr/security/RuleBasedAuthorizationPlugin.java#L65) null-safe.

After coding Kotlin for a few years I really miss its not-null-by-default!

Basically, this looks like a decent solution.

**Review by janhoy — APPROVED** [review]

**Comment by tflobbe** [comment]
> Add a CHANGES entry

Yes, I typically avoid adding an entry until I'm ready to merge, to prevent conflicts :). Added now.
